js/crm.drupal7.js

```diff
@@ -0,0 +1,44 @@
+(function($, crmMenubar) {
+  var numberOfMenusReady = 0;
+  var numberOfMenusToWaitFor = 2;
+
+  $(document).ready(runWhenMenusAreReady);
+  $(document).on('crmLoad', '#civicrm-menu', runWhenMenusAreReady);
+
+  /**
+   * Runs the hide menu toggle button when both the Admin Menu and the
+   * CiviCRM menu are ready.
+   */
+  function runWhenMenusAreReady() {
+    numberOfMenusReady++;
+
+    if (numberOfMenusReady < numberOfMenusToWaitFor) {
+      return;
+    }
+
+    hideMenuToggleButtonForNonAdminUsers();
+  }
```

Comment:
Working through the logic of this function, it looks complex but all it's really doing is [waiting for 2 events to fire](https://stackoverflow.com/questions/14892384/jquery-waiting-for-multiple-events-to-fire).